src/main/java/net/pms/dlna/DLNAProtocolInfo.java

```diff
@@ -0,0 +1,265 @@
+	 */
+	public DLNAProtocolInfo(ProtocolInfo template) {
+		this(template.getProtocol(),
+			template.getNetwork(),
+			template.getContentFormat(),
+			template.getAdditionalInfo()
+		);
+	}
+
+	/**
+	 * @return The {@link DLNAOrgFlags} for this {@link DLNAProtocolInfo}.
+	 */
+	public DLNAOrgFlags getFlags() {
+		return flags != null ? flags : DLNAOrgFlags.IMPLIED;
+	}
+
+	/**
+	 * @return The {@code DLNA.ORG_PN} (DLNA media format profile) for this
+	 *         {@link DLNAProtocolInfo} or {@code null} if none is defined.
+	 */
+	public String getPN() {
+		return pn;
+	}
+
+	/**
+	 * Creates a new {@link MimeType} from the content format of this
+	 * {@link DLNAProtocolInfo}.
+	 *
+	 * @return The corresponding {@link MimeType}.
+	 * @throws ParseException If {@code contentFormat} can't be parsed to a
+	 *             {@link MimeType}.
+	 */
+	public MimeType getMimeType() throws ParseException {
+		return MimeType.valueOf(contentFormat);
+	}
+
+	/**
+	 * Creates a new {@link org.seamless.util.MimeType} from the content format
+	 * of this {@link DLNAProtocolInfo}. To create a {@link MimeType}, use
+	 * {@link #getMimeType()} instead.
+	 *
+	 * @return The corresponding {@link org.seamless.util.MimeType}.
+	 * @see #getMimeType()
+	 */
+	@Override
+	public org.seamless.util.MimeType getContentFormatMimeType() throws IllegalArgumentException {
+		return org.seamless.util.MimeType.valueOf(contentFormat);
+	}
+
+	/**
+	 * Parses {@code additionalInfo} and stores the results.
+	 */
+	@Override
+	protected void parseAdditionalInfo() {
+		if (isBlank(additionalInfo)) {
+			return;
+		}
+		String[] parameters = additionalInfo.trim().toUpperCase(Locale.ROOT).split("\\s*;\\s*");
+		for (String parameter : parameters) {
+			if (isBlank(parameter)) {
+				continue;
+			}
+			String[] parameterEntry = parameter.split("\\s*=\\s*");
+			if (parameterEntry.length == 2) {
+				DLNAAttribute.Type type =
+					DLNAAttribute.Type.valueOfAttributeName(parameterEntry[0]);
+				if (type != null) {
+					DLNAAttribute<?> dlnaAttrinute;
+					try {
+						dlnaAttrinute = DLNAAttribute.newInstance(type, parameterEntry[1], getMimeType().toStringWithoutParameters());
+					} catch (ParseException e) {
+						dlnaAttrinute = DLNAAttribute.newInstance(type, parameterEntry[1], null);
+						LOGGER.debug("Error parsing mime-type while parsing additionalInfo from \"{}\": {}", this, e.getMessage());
+						LOGGER.trace("", e);
+					}
+					if (dlnaAttrinute != null) {
+						attributes.put(type, dlnaAttrinute);
+					}
+					switch (type) {
+						case DLNA_ORG_FLAGS:
+							flags = new DLNAOrgFlags(parameterEntry[1]);
+							break;
+						case DLNA_ORG_PN:
+							pn = parameterEntry[1];
+							break;
+						case DLNA_ORG_CI:
+						case DLNA_ORG_OP:
+						case DLNA_ORG_PS:
+						default:
+							break;
+					}
+				} else if (!"SONY.COM_PN".equals(parameterEntry[0])) {
+					LOGGER.debug("Unknown ProtocolInfo parameter type \"{}\" ({})", parameterEntry[0], parameter);
```

Comment:
@Sami32 There was some problems with the parsing that set me on the current path. I had tried to take some "shortcuts" by using some parts from Cling, but it turns out that their implementation isn't ideal (IMO). So, now I'm replacing most of that with "my own" implementation that behaves the way I want, which will allow for a much more general solution.

I've actually completed most of it, but I'm currently stuck on something really stupid, and that is implementing a proper ```Rational``` number type. The reason is that the ```PS``` parameters states that the values must be exact. Programming languages generally doesn't do "exact" non-integer values, floating point (```float```, ```double```) is the most common. But, they have some serious rounding issues, and can't even store a simple value like "0.1" exactly. Thus, I figured that a ```Rational``` type would be the way to go. I found some existing implementations, but none was exactly as I wanted - so now I'm trying to make it myself. 

As usual this goes far beyond anything reasonable for this PR, but I'm thinking that when I first make a ```Rational``` type I just as well make it as versatile as possible, so that it can be used generally and not just implemented for the relatively simple needs of the ```PS``` parameter.

It's not complicated to implement either, were it not for all the limitations of the existing number types. There are so many corner cases to handle, bits to keep track of and overflows to avoid that this has taken a lot longer than I intended....